mpi_method_server.py

```diff
@@ -0,0 +1,94 @@
+import argparse
+from multiprocessing import Queue
+import redis
+import time
+import os
+
+import parsl
+from parsl import python_app, bash_app
+from parsl.executors import ThreadPoolExecutor
+from parsl.executors import HighThroughputExecutor
+from parsl.providers import LocalProvider
+from parsl.config import Config
+from parsl.data_provider.files import File
+from concurrent.futures import Future
+
+from redis_q import RedisQueue
+
+class MpiMethodServer:
+
+    def __init__(self, input_queue, output_queue):
+        self.input_queue  = input_queue
+        self.output_queue = output_queue
+        self.task_list    = []
+
+    # Simulate will run some commands on bash, this can be made to run MPI applications via aprun
+    # Here, simulate will put a random number from range(0-32767) into the output file.
+    # ************ NOTE: Seems that a bash_app cannot take a "self", so I need to remove
```

Comment:
If simulate took self as a parameter, then at call time the MpiMethodServer object would need to be shipped to the remote context. Since class objects often hold mutable state, this is unsafe in a concurrent execution environment. One alternative would be to have a `MpiMethodServer.add_method` call that adds a parsl_app dynamically into a dict of functions. Separating out the methods from the class would keep them as pure functions, and modular.

Reply:
I figured that it was something like that. I understand what you are saying, but am not clear what is to be done. Can you show me how to do this?

Reply:
Here's a quick snippet : https://gist.github.com/yadudoc/b19c8b7821068ab510fd9b317d748774